### ISSUE TITLE
In debug mode `findfiles2` could print out why it skipped a particular file that matched the pattern

### ISSUE THREAD

**Issue by karthiknadig** [microsoft/vscode#205762]
Testing #205408

When testing, I had a folder with `.gitignore` (content: `*`) which self ignores the entire folder. It was confusing to figure out why. If I knew it was getting skipped due to ignore it would have helped. In `debugpy` we have similar rules, and we describe why a file was ignored or skipped for step-into in logs when set to verbose.

**Comment by andreamah** [comment]
@jrieken I'm curious if other APIs have a `debug` flag that allows them to show more debug info. We could add debug/trace logs when you search for files, but I'd assume that it gets annoying after a while.

**Comment by jrieken** [comment]
Having this would be help but it is nothing that should be in the API. I believe trace/debug logging is the way to go here but I also wonder if RG gives you all this information?

**Comment by andreamah** [comment]
That's true. When things are excluded at the RG level, it actually doesn't list the excludes that it gets from parsing ignore files, so it'd be hard to get specific files in this case also.

I wonder whether it'd be worth it to just have a trace or debug log that shows which settings that it uses and which fields are fixed. For example
```
- using files.exclude setting -> false
- using search.exclude setting -> false
- using useIgnoreFiles -> following settings -> true (the value from settings)
- using useGlobalIgnore-> following settings -> false (the value from settings)
- using useParentIgnore-> following settings -> false (the value from settings)
```

This is a lot easier to show than all of the globs that it finds (since some of it is done by ripgrep and we'd otherwise need to make an extra call perhaps to list exclude globs). It would be even harder to list all ignored files.

@karthiknadig Do you think that this would be sufficient in this case? I recall that a bit issue was that it was unclear what `useIgnoreFiles` actually resolved to.

**Comment by karthiknadig** [comment]
Yes. In my case my confusion stemmed from the fact that I had not set `useIgnoreFiles` and my assumption was not setting it is same as `false`. The above logs would have helped in that case.
